### PR TITLE
[UI] Cleanup App and Game Settings

### PR DESCRIPTION
**Pull request by flavioislima**
This is the first part of refactoring the Settings.
It removes settings that are broken or not necessary anymore and adjust certain settings depending on the runner since not all of them affects HyperPlay Games

### Changes

- Removed Anti-cheat settings for HyperPlay Games. Reason: Won't work.
- Removed `AltGOGDL` and `AltLegendaryEXE` advanced settings Reason: Not useful
- Removed "Ignore Game Updates" setting for HyperPlay games: Reason, all games should be on the latest version.
- Commented out for now the Disable Controller settings since controller navigation is broken since updating to Mantine-UI.

### How to Test

Check if the settings were properly removed and if the Anti-Cheat Game settings only appears on Epic and GOG Games (should still appear on the Main Game settings on HyperPlay Settings).

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by flavioislima** [comment]
Let me know if we should keep any of those settings and if there are others we could remove or hide due to some condition.

File: src/frontend/screens/Library/components/InstallModal/DownloadDialog/index.tsx
[the remark this answers is not in the document]
Comment (flavioislima) [reply src/frontend/screens/Library/components/InstallModal/DownloadDialog/index.tsx:1]: Yes, I did it here before to make all inputs use the TextInput from HP-UI.
I will fix the conflicts once that gets merged.

[the remark this answers is not in the document]
Comment (flavioislima) [reply src/frontend/screens/Library/components/InstallModal/DownloadDialog/index.tsx:1]: better, just reverted the commit with this change.

**Review by BrettCleary — APPROVED** [review]
LGTM

**Review by eliobricenov — APPROVED** [review]

**Pull request merged**